### PR TITLE
check for openhab environent and php support.

### PR DESCRIPTION
**Pull request by peuter**
This is a cherry-picked feature taken from a future changeset for the development branch. This is the minimum requirement to make the cometvisu work when its served by the openHAB server. The check is needed because some of the backend URLs are different in openHAB.

### CODE REVIEW

File: source/class/cv/io/rest/Client.js [hunk 1]
```diff
@@ -39,10 +39,8 @@ qx.Class.define('cv.io.rest.Client', {
     getBaseUrl: function () {
       if (!this.BASE_URL) {
         let path = '';
-        const engine = cv.TemplateEngine.getInstance();
-        const clientBackend = engine.visu && typeof engine.visu.getBackend === 'function' ? engine.visu.getBackend() : {};
-        if (clientBackend.resources && clientBackend.resources.rest) {
-          path = clientBackend.resources.rest;
```
Comment (ChristianMayer) [inline source/class/cv/io/rest/Client.js:45]: Is this code path only relevant for OpenHAB?
Or could it also be used by KNX, e.g. when served by the Docker container?

Comment (peuter) [reply source/class/cv/io/rest/Client.js:45]: This has always been a mix-up of two different things. The URL of the CometVisus REST-API has nothing to do with the backend used for state handling. e.g. you can use openHAB as "state-backend" and still have the normal PHP-REST-API (when you use the docker container). Only if you serve the CometVisu via openHABs own webserver, you use the REST-API that the cometvisu binding provides (same features as the PHP stuff, but natively implemented in java to make it work in openhab). And fpr that you need a slightly different URL. And thats what this is all about. So you only need to know if the CometVisu is served via openHAB.

**Comment by ChristianMayer** [comment]
Opening the demo config with this PR included creates this error message on the console:
grafik

**Comment by ChristianMayer** [comment]
Now I tried a plain `release-0.12` branch without this PR - and it also shows this message.
But I didn't switch back to Qx 6 for that.

Could that be the cause of this message?

**Comment by peuter** [comment]
> Could that be the cause of this message?

You never know, but I don't think so. I see the same request here, but it gets an answer and is working just fine. As the error show this is an server error (Status 500) you will find an error in you server logs that will describe the problem.
But it has nothing to do with this pull request.

**Review by ChristianMayer — APPROVED** [review]

**Pull request merged**